here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/executors/write/UpdateFeature.kt

```diff
@@ -20,23 +20,21 @@ class UpdateFeature(
     //TODO this implementation currently do not support atomic updates!
     // In other words, it ignores if version is set in the Write operation,
     // which requires that the current HEAD state is exactly in this version.
-    fun execute(collection: PgCollection, write: WriteExt): Tuple {
+    fun execute(collection: PgCollection, write: WriteExt): Triple<Tuple?, String, String> {
```

Comment:
Please use exception handling for error handling, not arbitrary string values. You can throw an NakshaException, and catch it in the calling method, I do not see why this Triple is of any use here?